diff --git a/create-react-app-typescript/cypress.json b/create-react-app-typescript/cypress.json
new file mode 100644
index 0000000..ee59bec
--- /dev/null
+++ b/create-react-app-typescript/cypress.json
@@ -0,0 +1,6 @@
+{
+  "component": {
+    "testFiles": "**/*.spec.ct.{js,ts,jsx,tsx}",
+    "componentFolder": "src"
+  }
+}
diff --git a/create-react-app-typescript/cypress/fixtures/example.json b/create-react-app-typescript/cypress/fixtures/example.json
new file mode 100644
index 0000000..02e4254
--- /dev/null
+++ b/create-react-app-typescript/cypress/fixtures/example.json
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
diff --git a/create-react-app-typescript/cypress/plugins/index.ts b/create-react-app-typescript/cypress/plugins/index.ts
new file mode 100644
index 0000000..5398153
--- /dev/null
+++ b/create-react-app-typescript/cypress/plugins/index.ts
@@ -0,0 +1,6 @@
+const injectDevServer = require("@cypress/react/plugins/react-scripts");
+
+module.exports = (on, config) => {
+  injectDevServer(on, config);
+  return config;
+};
diff --git a/create-react-app-typescript/cypress/support/commands.ts b/create-react-app-typescript/cypress/support/commands.ts
new file mode 100644
index 0000000..119ab03
--- /dev/null
+++ b/create-react-app-typescript/cypress/support/commands.ts
@@ -0,0 +1,25 @@
+// ***********************************************
+// This example commands.js shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add('login', (email, password) => { ... })
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
diff --git a/create-react-app-typescript/cypress/support/index.ts b/create-react-app-typescript/cypress/support/index.ts
new file mode 100644
index 0000000..d68db96
--- /dev/null
+++ b/create-react-app-typescript/cypress/support/index.ts
@@ -0,0 +1,20 @@
+// ***********************************************************
+// This example support/index.js is processed and
+// loaded automatically before your test files.
+//
+// This is a great place to put global configuration and
+// behavior that modifies Cypress.
+//
+// You can change the location of this file or turn off
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
diff --git a/create-react-app-typescript/cypress/tsconfig.json b/create-react-app-typescript/cypress/tsconfig.json
new file mode 100644
index 0000000..78f2e27
--- /dev/null
+++ b/create-react-app-typescript/cypress/tsconfig.json
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "lib": ["es5", "dom"],
+    "types": ["cypress", "node"]
+  },
+  "include": ["**/*.ts", "**/*.tsx"]
+}
