diff --git a/vue-cli-3/.gitignore b/vue-cli-3/.gitignore
index 403adbc..f6061b7 100644
--- a/vue-cli-3/.gitignore
+++ b/vue-cli-3/.gitignore
@@ -2,6 +2,8 @@
 node_modules
 /dist
 
+.nyc_output
+coverage
 
 # local env files
 .env.local
diff --git a/vue-cli-3/.nycrc b/vue-cli-3/.nycrc
new file mode 100644
index 0000000..3a40283
--- /dev/null
+++ b/vue-cli-3/.nycrc
@@ -0,0 +1,13 @@
+{
+  "include": [
+    "src/**/*.{js,vue}"
+  ],
+  "exclude": [
+    "src/**/*.spec.ct.js"
+  ],
+  "extension": [
+    ".js",
+    ".vue"
+  ],
+  "all": true
+}
\ No newline at end of file
diff --git a/vue-cli-3/babel.config.js b/vue-cli-3/babel.config.js
index e955840..9f7fe23 100644
--- a/vue-cli-3/babel.config.js
+++ b/vue-cli-3/babel.config.js
@@ -1,5 +1,12 @@
+const plugins = []
+
+if (process.env.CYPRESS) {
+  plugins.push('istanbul')
+}
+
 module.exports = {
   presets: [
     '@vue/cli-plugin-babel/preset'
-  ]
-}
+  ],
+  plugins
+}
\ No newline at end of file
diff --git a/vue-cli-3/cypress.json b/vue-cli-3/cypress.json
index ee59bec..622829d 100644
--- a/vue-cli-3/cypress.json
+++ b/vue-cli-3/cypress.json
@@ -1,4 +1,5 @@
 {
+  "coverage": true,
   "component": {
     "testFiles": "**/*.spec.ct.{js,ts,jsx,tsx}",
     "componentFolder": "src"
diff --git a/vue-cli-3/cypress/plugins/index.js b/vue-cli-3/cypress/plugins/index.js
index 3590bd9..1f08a26 100644
--- a/vue-cli-3/cypress/plugins/index.js
+++ b/vue-cli-3/cypress/plugins/index.js
@@ -1,5 +1,6 @@
 const { startDevServer } = require('@cypress/webpack-dev-server')
 const webpackConfig = require('@vue/cli-service/webpack.config')
+const codeCoverageTask = require('@cypress/code-coverage/task')
 
 module.exports = (on, config) => {
   on('dev-server:start', (options) => {
@@ -8,5 +9,7 @@ module.exports = (on, config) => {
       webpackConfig,
     })
   })
+
+  codeCoverageTask(on, config)
   return config
 }
diff --git a/vue-cli-3/cypress/support/index.js b/vue-cli-3/cypress/support/index.js
index 9b7843f..05cf3eb 100644
--- a/vue-cli-3/cypress/support/index.js
+++ b/vue-cli-3/cypress/support/index.js
@@ -13,6 +13,8 @@
 // https://on.cypress.io/configuration
 // ***********************************************************
 
+import '@cypress/code-coverage/support'
+
 // Import commands.js using ES2015 syntax:
 import './commands'
