diff --git a/create-react-app-typescript/package.json b/create-react-app-typescript/package.json
index b12408e..5343d42 100644
--- a/create-react-app-typescript/package.json
+++ b/create-react-app-typescript/package.json
@@ -25,7 +25,8 @@
   "eslintConfig": {
     "extends": [
       "react-app",
-      "react-app/jest"
+      "react-app/jest",
+      "plugin:cypress/recommended"
     ]
   },
   "browserslist": {
diff --git a/create-react-app-typescript/src/App.cy.tsx b/create-react-app-typescript/src/App.cy.tsx
new file mode 100644
index 0000000..21daa26
--- /dev/null
+++ b/create-react-app-typescript/src/App.cy.tsx
@@ -0,0 +1,8 @@
+import React from 'react';
+import { mount } from '@cypress/react';
+import App from './App';
+
+it('renders learn react link', () => {
+  mount(<App />);
+  cy.get('a').contains('Learn React');
+});
diff --git a/setup-create-react-app-typescript/src/App.test.tsx b/setup-create-react-app-typescript/src/App.test.tsx
index 2a68616..6d40586 100644
--- a/setup-create-react-app-typescript/src/App.test.tsx
+++ b/setup-create-react-app-typescript/src/App.test.tsx
@@ -5,5 +5,6 @@ import App from './App';
 test('renders learn react link', () => {
   render(<App />);
   const linkElement = screen.getByText(/learn react/i);
+  // @ts-ignore
   expect(linkElement).toBeInTheDocument();
 });
