diff --git a/setup-vue-cli-3/cypress.config.js b/setup-vue-cli-3/cypress.config.js
new file mode 100644
index 0000000..eabdc78
--- /dev/null
+++ b/setup-vue-cli-3/cypress.config.js
@@ -0,0 +1,11 @@
+const { defineConfig } = require('cypress')
+const { devServer } = require('@cypress/webpack-dev-server')
+const webpackConfig = require('@vue/cli-service/webpack.config')
+
+module.exports = defineConfig({
+  // Component testing, JavaScript, Vue CLI, Webpack
+  component: {
+    devServer,
+    devServerConfig: { webpackConfig }
+  },
+})
\ No newline at end of file
diff --git a/setup-vue-cli-3/cypress/fixtures/example.json b/setup-vue-cli-3/cypress/fixtures/example.json
new file mode 100644
index 0000000..02e4254
--- /dev/null
+++ b/setup-vue-cli-3/cypress/fixtures/example.json
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
diff --git a/setup-vue-cli-3/cypress/component/index.html b/setup-vue-cli-3/cypress/component/index.html
new file mode 100644
index 0000000..410b588
--- /dev/null
+++ b/setup-vue-cli-3/cypress/component/index.html
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>Components App</title>
+
+  </head>
+  <body>
+
+    <div id="__cy_root"></div>
+  </body>
+</html>
\ No newline at end of file
diff --git a/setup-vue-cli-3/cypress/support/commands.js b/setup-vue-cli-3/cypress/support/commands.js
new file mode 100644
index 0000000..119ab03
--- /dev/null
+++ b/setup-vue-cli-3/cypress/support/commands.js
@@ -0,0 +1,25 @@
+// ***********************************************
+// This example commands.js shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add('login', (email, password) => { ... })
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
diff --git a/setup-vue-cli-2/cypress/support/component.js b/setup-vue-cli-2/cypress/support/component.js
new file mode 100644
index 0000000..7e654cd
--- /dev/null
+++ b/setup-vue-cli-2/cypress/support/component.js
@@ -0,0 +1,5 @@
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Ensure global app styles are loaded:
+import '../../src/main.css'
\ No newline at end of file
diff --git a/setup-vue-cli-3/src/App.cy.js b/setup-vue-cli-3/src/App.cy.js
new file mode 100644
index 0000000..4a912ba
--- /dev/null
+++ b/setup-vue-cli-3/src/App.cy.js
@@ -0,0 +1,11 @@
+import { mount } from '@cypress/vue'
+import App from './App.vue'
+
+it('shows the App', () => {
+  mount(App)
+  cy.get('[alt="Vue logo"]')
+  cy.get('h1').contains('Welcome to Your Vue.js App')
+  cy.get('button').click()
+  cy.get('[data-test=count]').should("have.text", "Total clicks: 1")
+  cy.get('button').click()
+})
diff --git a/setup-vue-cli-3/src/components/HelloWorld.cy.js b/setup-vue-cli-3/src/components/HelloWorld.cy.js
new file mode 100644
index 0000000..bbbb6c3
--- /dev/null
+++ b/setup-vue-cli-3/src/components/HelloWorld.cy.js
@@ -0,0 +1,11 @@
+import { mount } from '@cypress/vue'
+import HelloWorld from './HelloWorld.vue'
+
+it('shows the header', () => {
+  mount(HelloWorld, {
+    propsData: {
+      msg: 'Hello World'
+    }
+  })
+  cy.get('h1').contains('Hello World')
+})
diff --git a/vue-cli-3/src/App.vue b/vue-cli-3/src/App.vue
index a92fd03..3d9f207 100644
--- a/vue-cli-3/src/App.vue
+++ b/vue-cli-3/src/App.vue
@@ -1,15 +1,20 @@
 <template>
-  <img alt="Vue logo" src="./assets/logo.png">
-  <HelloWorld msg="Welcome to Your Vue.js App"/>
+  <div>
+    <img alt="Vue logo" src="./assets/logo.png">
+    <HelloWorld msg="Welcome to Your Vue.js App"/>
+    <Counter />
+  </div>
 </template>

 <script>
 import HelloWorld from './components/HelloWorld.vue'
+import Counter from './components/Counter.vue'

 export default {
   name: 'App',
   components: {
-    HelloWorld
+    HelloWorld,
+    Counter
   }
 }
 </script>
diff --git a/vue-cli-3/src/components/Counter.vue b/vue-cli-3/src/components/Counter.vue
new file mode 100644
index 0000000..5eb6269
--- /dev/null
+++ b/vue-cli-3/src/components/Counter.vue
@@ -0,0 +1,20 @@
+<template>
+  <div>
+    <p data-test="count">Total clicks: {{ count }}</p>
+    <button @click="increment">increment</button>
+  </div>
+</template>
+
+<script>
+export default {
+  data: () => ({
+    count: 0,
+  }),
+
+  methods: {
+    increment() {
+      this.count++
+    },
+  },
+}
+</script>
\ No newline at end of file
diff --git a/vue-cli-3/src/components/Counter.cy.js b/vue-cli-3/src/components/Counter.cy.js
new file mode 100644
index 0000000..afd7e21
--- /dev/null
+++ b/vue-cli-3/src/components/Counter.cy.js
@@ -0,0 +1,10 @@
+import { mount } from '@cypress/vue'
+import Counter from './Counter.vue'
+
+it('shows the Counter', () => {
+  mount(Counter)
+  cy.get('button').click()
+  cy.get('[data-test=count]').should("have.text", "Total clicks: 1")
+  cy.get('button').click()
+  cy.get('[data-test=count]').should("have.text", "Total clicks: 2")
+})
diff --git a/vue-cli-3/src/components/CounterCompositionApi.cy.js b/vue-cli-3/src/components/CounterCompositionApi.cy.js
new file mode 100644
index 0000000..afd7e21
--- /dev/null
+++ b/vue-cli-3/src/components/CounterCompositionApi.cy.js
@@ -0,0 +1,10 @@
+import { mount } from '@cypress/vue'
+import CounterCompositionApi from './CounterCompositionApi.vue'
+
+it('shows the CounterCompositionApi', () => {
+  mount(CounterCompositionApi)
+  cy.get('button').click()
+  cy.get('[data-test=count]').should("have.text", "Total clicks: 1")
+  cy.get('button').click()
+  cy.get('[data-test=count]').should("have.text", "Total clicks: 2")
+})
diff --git a/vue-cli-3/src/components/CounterCompositionApi.vue b/vue-cli-3/src/components/CounterCompositionApi.vue
new file mode 100644
index 0000000..bb58451
--- /dev/null
+++ b/vue-cli-3/src/components/CounterCompositionApi.vue
@@ -0,0 +1,19 @@
+<template>
+  <div>
+    <button @click="increment">Increment</button>
+    <p data-test="count">Total clicks: {{ count }}</p>
+  </div>
+</template>
+
+<script>
+import { ref } from 'vue'
+
+export default {
+  setup() {
+    const count = ref(0)
+    const increment = () => ++count.value
+
+    return { count, increment }
+  }
+}
+</script>
\ No newline at end of file
