diff --git a/setup-create-react-app-redux/src/app/store.js b/setup-create-react-app-redux/src/app/store.js
index 9eca6d2..57b1c0c 100644
--- a/setup-create-react-app-redux/src/app/store.js
+++ b/setup-create-react-app-redux/src/app/store.js
@@ -1,8 +1,10 @@
 import { configureStore } from '@reduxjs/toolkit';
 import counterReducer from '../features/counter/counterSlice';
 
-export const store = configureStore({
+export const getStore = () => configureStore({
   reducer: {
     counter: counterReducer,
   },
 });
+
+export const store = getStore();
