diff --git a/vite-react/src/App.css b/vite-react/src/App.css
index 8da3fde..614b3de 100644
--- a/vite-react/src/App.css
+++ b/vite-react/src/App.css
@@ -14,7 +14,6 @@
 }
 
 .App-header {
-  background-color: #282c34;
   min-height: 100vh;
   display: flex;
   flex-direction: column;
@@ -36,7 +35,3 @@
     transform: rotate(360deg);
   }
 }
-
-button {
-  font-size: calc(10px + 2vmin);
-}
diff --git a/vite-react/src/App.jsx b/vite-react/src/App.jsx
index 0939547..ff835a2 100644
--- a/vite-react/src/App.jsx
+++ b/vite-react/src/App.jsx
@@ -1,19 +1,15 @@
-import { useState } from 'react'
 import logo from './logo.svg'
+import { Counter } from './Counter'
 import './App.css'
 
 function App() {
-  const [count, setCount] = useState(0)
-
   return (
     <div className="App">
       <header className="App-header">
         <img src={logo} className="App-logo" alt="logo" />
         <p>Hello Vite + React!</p>
         <p>
-          <button type="button" onClick={() => setCount((count) => count + 1)}>
-            count is: {count}
-          </button>
+          <Counter />
         </p>
         <p>
           Edit <code>App.jsx</code> and save to test HMR updates.
diff --git a/vite-react/src/Counter.css b/vite-react/src/Counter.css
new file mode 100644
index 0000000..42dbfa0
--- /dev/null
+++ b/vite-react/src/Counter.css
@@ -0,0 +1,3 @@
+button {
+  font-size: calc(10px + 2vmin);
+}
diff --git a/vite-react/src/Counter.jsx b/vite-react/src/Counter.jsx
new file mode 100644
index 0000000..6990784
--- /dev/null
+++ b/vite-react/src/Counter.jsx
@@ -0,0 +1,12 @@
+import { useState } from 'react'
+import './Counter.css'
+
+export const Counter = () => {
+  const [count, setCount] = useState(0)
+
+  return (
+    <button type="button" onClick={() => setCount((count) => count + 1)}>
+      count is: {count}
+    </button>
+  )
+}
diff --git a/vite-react/src/index.css b/vite-react/src/index.css
index ec2585e..af97127 100644
--- a/vite-react/src/index.css
+++ b/vite-react/src/index.css
@@ -1,5 +1,6 @@
 body {
   margin: 0;
+  background-color: #282c34;
   font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
     'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
     sans-serif;
