diff --git a/setup-create-react-app-react-router/package.json b/setup-create-react-app-react-router/package.json
index 2dfdaa4..c341cb6 100644
--- a/setup-create-react-app-react-router/package.json
+++ b/setup-create-react-app-react-router/package.json
@@ -20,7 +20,8 @@
   "eslintConfig": {
     "extends": [
       "react-app",
-      "react-app/jest"
+      "react-app/jest",
+      "plugin:cypress/recommended"
     ]
   },
   "browserslist": {
diff --git a/setup-create-react-app-react-router/src/App.cy.js b/setup-create-react-app-react-router/src/App.cy.js
new file mode 100644
index 0000000..59b4cf8
--- /dev/null
+++ b/setup-create-react-app-react-router/src/App.cy.js
@@ -0,0 +1,31 @@
+import { mount } from "@cypress/react";
+import { MemoryRouter } from "react-router";
+import App from "./App";
+
+const TestApp = ({ path = "/", initialEntries = [path] }) => (
+  <MemoryRouter initialEntries={initialEntries}>
+    <App />
+  </MemoryRouter>
+);
+
+it("renders the home page when the path is /", () => {
+  mount(<TestApp path="/" />);
+  cy.get(".Page").contains("Home page content");
+});
+
+it("navigates to the about page", () => {
+  mount(<TestApp path="/" />);
+  cy.get("a").contains("About").click();
+  cy.get(".Page").contains("About page content");
+});
+
+it("renders the about page when the path is /about", () => {
+  mount(<TestApp path="/about" />);
+  cy.get(".Page").contains("About page content");
+});
+
+it("navigates to the home page", () => {
+  mount(<TestApp path="/about" />);
+  cy.get("a").contains("Home").click();
+  cy.get(".Page").contains("Home page content");
+});
