diff --git a/setup-vite-vue/.gitignore b/setup-vite-vue/.gitignore
index a547bf3..e98cec5 100644
--- a/setup-vite-vue/.gitignore
+++ b/setup-vite-vue/.gitignore
@@ -11,6 +11,8 @@ node_modules
 dist
 dist-ssr
 *.local
+.nyc_output
+coverage

 # Editor directories and files
 .vscode/*
diff --git a/setup-vite-vue/cypress.json b/setup-vite-vue/cypress.json
index ab25fd8..e44ce02 100644
--- a/setup-vite-vue/cypress.json
+++ b/setup-vite-vue/cypress.json
@@ -1,4 +1,5 @@
 {
+  "coverage": true,
   "component": {
     "testFiles": "**/*.spec.ct.{js,ts,jsx,tsx}",
     "componentFolder": "src"
diff --git a/setup-vite-vue/cypress/plugins/index.js b/setup-vite-vue/cypress/plugins/index.js
index 523f1fd..e111f19 100644
--- a/setup-vite-vue/cypress/plugins/index.js
+++ b/setup-vite-vue/cypress/plugins/index.js
@@ -1,15 +1,20 @@
 const path = require('path')
 const { startDevServer } = require('@cypress/vite-dev-server')
+const codeCoverageTask = require('@cypress/code-coverage/task')
+const istanbul = require('vite-plugin-istanbul')
 
 module.exports = (on, config) => {
   on('dev-server:start', (options) => {
     return startDevServer({
       options,
       viteConfig: {
-        configFile: path.resolve(__dirname, '../../vite.config.js'),
+        clearScreen: false,
+        plugins: [istanbul({})],
       },
     })
   })
 
+  codeCoverageTask(on, config);
+
   return config
 }
diff --git a/setup-vite-vue/cypress/support/index.js b/setup-vite-vue/cypress/support/index.js
index 9b7843f..05cf3eb 100644
--- a/setup-vite-vue/cypress/support/index.js
+++ b/setup-vite-vue/cypress/support/index.js
@@ -13,6 +13,8 @@
 // https://on.cypress.io/configuration
 // ***********************************************************

+import '@cypress/code-coverage/support'
+
 // Import commands.js using ES2015 syntax:
 import './commands'
