diff --git a/create-react-app-tailwind/cypress/support/index.js b/create-react-app-tailwind/cypress/support/index.js
index c63e2c9..e7a097d 100644
--- a/create-react-app-tailwind/cypress/support/index.js
+++ b/create-react-app-tailwind/cypress/support/index.js
@@ -19,5 +19,8 @@ import './commands'
 // Alternatively you can use CommonJS syntax:
 // require('./commands')
 
+// Ensure tailwind styles are loaded:
+import 'tailwindcss/dist/tailwind.css'
+
 // Ensure global app styles are loaded:
 import '../../src/index.css'
diff --git a/create-react-app-tailwind/package.json b/create-react-app-tailwind/package.json
index 9c064e8..7c968d6 100644
--- a/create-react-app-tailwind/package.json
+++ b/create-react-app-tailwind/package.json
@@ -21,7 +21,8 @@
   "eslintConfig": {
     "extends": [
       "react-app",
-      "react-app/jest"
+      "react-app/jest",
+      "plugin:cypress/recommended"
     ]
   },
   "browserslist": {
diff --git a/create-react-app-tailwind/src/App.spec.ct.js b/create-react-app-tailwind/src/App.spec.ct.js
new file mode 100644
index 0000000..21daa26
--- /dev/null
+++ b/create-react-app-tailwind/src/App.spec.ct.js
@@ -0,0 +1,8 @@
+import React from 'react';
+import { mount } from '@cypress/react';
+import App from './App';
+
+it('renders learn react link', () => {
+  mount(<App />);
+  cy.get('a').contains('Learn React');
+});
