diff --git a/create-nuxt-app/components/Logo.spec.ct.js b/create-nuxt-app/components/Logo.spec.ct.js
new file mode 100644
index 0000000..4f160f8
--- /dev/null
+++ b/create-nuxt-app/components/Logo.spec.ct.js
@@ -0,0 +1,7 @@
+import { mount } from '@cypress/vue'
+import NuxtLogo from './NuxtLogo.vue';
+
+it('shows the logo', () => {
+  mount(NuxtLogo)
+  cy.get('svg').should('have.class', 'nuxt-logo');
+})
diff --git a/create-nuxt-app/components/Counter.vue b/create-nuxt-app/components/Counter.vue
new file mode 100644
index 0000000..bdcd69f
--- /dev/null
+++ b/create-nuxt-app/components/Counter.vue
@@ -0,0 +1,19 @@
+<template>
+  <div>
+    <p data-test="count">Total clicks: {{ count }}</p>
+    <button @click="increment">increment</button>
+  </div>
+</template>
+
+<script>
+export default {
+  data: () => ({
+    count: 0,
+  }),
+  methods: {
+    increment() {
+      this.count++
+    },
+  },
+}
+</script>
\ No newline at end of file
diff --git a/create-nuxt-app/pages/index.vue b/create-nuxt-app/pages/index.vue
index 8360af7..c7b8627 100644
--- a/create-nuxt-app/pages/index.vue
+++ b/create-nuxt-app/pages/index.vue
@@ -1,7 +1,18 @@
 <template>
-  <Tutorial/>
+  <div>
+    <Counter />
+    <Tutorial />
+  </div>
 </template>

 <script>
-export default {}
+import Tutorial from '../components/Tutorial.vue'
+import Counter from '../components/Counter.vue'
+
+export default {
+  components: {
+    Counter,
+    Tutorial
+  }
+}
 </script>
diff --git a/create-nuxt-app/cypress.json b/create-nuxt-app/cypress.json
new file mode 100644
index 0000000..a57b21f
--- /dev/null
+++ b/create-nuxt-app/cypress.json
@@ -0,0 +1,6 @@
+{
+  "component": {
+    "testFiles": "**/*.spec.ct.{js,ts,jsx,tsx}",
+    "componentFolder": "."
+  }
+}
diff --git a/create-nuxt-app/cypress/fixtures/example.json b/create-nuxt-app/cypress/fixtures/example.json
new file mode 100644
index 0000000..02e4254
--- /dev/null
+++ b/create-nuxt-app/cypress/fixtures/example.json
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
diff --git a/create-nuxt-app/cypress/plugins/index.js b/create-nuxt-app/cypress/plugins/index.js
new file mode 100644
index 0000000..ba79b66
--- /dev/null
+++ b/create-nuxt-app/cypress/plugins/index.js
@@ -0,0 +1,13 @@
+const { startDevServer } = require('@cypress/webpack-dev-server')
+const { getWebpackConfig } = require('nuxt')
+
+module.exports = (on, config) => {
+  on('dev-server:start', async (options) => {
+    const webpackConfig = await getWebpackConfig()
+    return startDevServer({
+      options,
+      webpackConfig,
+    })
+  })
+  return config
+}
diff --git a/create-nuxt-app/cypress/support/commands.js b/create-nuxt-app/cypress/support/commands.js
new file mode 100644
index 0000000..119ab03
--- /dev/null
+++ b/create-nuxt-app/cypress/support/commands.js
@@ -0,0 +1,25 @@
+// ***********************************************
+// This example commands.js shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add('login', (email, password) => { ... })
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
diff --git a/create-nuxt-app/cypress/support/index.js b/create-nuxt-app/cypress/support/index.js
new file mode 100644
index 0000000..d68db96
--- /dev/null
+++ b/create-nuxt-app/cypress/support/index.js
@@ -0,0 +1,20 @@
+// ***********************************************************
+// This example support/index.js is processed and
+// loaded automatically before your test files.
+//
+// This is a great place to put global configuration and
+// behavior that modifies Cypress.
+//
+// You can change the location of this file or turn off
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
diff --git a/create-nuxt-app/pages/index.spec.ct.js b/create-nuxt-app/pages/index.spec.ct.js
new file mode 100644
index 0000000..02161f9
--- /dev/null
+++ b/create-nuxt-app/pages/index.spec.ct.js
@@ -0,0 +1,10 @@
+import { mount } from '@cypress/vue'
+import Index from './index.vue'
+
+it('shows the index', () => {
+  mount(Index)
+  cy.get("h2").should("contain", "Welcome to your Nuxt Application");
+  cy.get("button").click();
+  cy.get("[data-test=count]").should("have.text", "Total clicks: 1");
+  cy.get("button").click();
+})
