diff --git a/create-react-app-tailwind/craco.config.js b/create-react-app-tailwind/craco.config.js
new file mode 100644
index 0000000..6a495d7
--- /dev/null
+++ b/create-react-app-tailwind/craco.config.js
@@ -0,0 +1,10 @@
+module.exports = {
+  style: {
+    postcss: {
+      plugins: [
+        require('tailwindcss'),
+        require('autoprefixer'),
+      ],
+    },
+  },
+}
diff --git a/create-react-app-tailwind/package.json b/create-react-app-tailwind/package.json
index 83f5f4e..64122c7 100644
--- a/create-react-app-tailwind/package.json
+++ b/create-react-app-tailwind/package.json
@@ -13,9 +13,9 @@
     "web-vitals": "^1.0.1"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
+    "start": "craco start",
+    "build": "craco build",
+    "test": "craco test",
     "eject": "react-scripts eject"
   },
   "eslintConfig": {
diff --git a/create-react-app-tailwind/src/index.css b/create-react-app-tailwind/src/index.css
index ec2585e..17df0e7 100644
--- a/create-react-app-tailwind/src/index.css
+++ b/create-react-app-tailwind/src/index.css
@@ -1,3 +1,7 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
 body {
   margin: 0;
   font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
diff --git a/create-react-app-tailwind/tailwind.config.js b/create-react-app-tailwind/tailwind.config.js
new file mode 100644
index 0000000..7b39079
--- /dev/null
+++ b/create-react-app-tailwind/tailwind.config.js
@@ -0,0 +1,11 @@
+module.exports = {
+  purge: ['./src/**/*.{js,jsx,ts,tsx}', './public/index.html'],
+  darkMode: false, // or 'media' or 'class'
+  theme: {
+    extend: {},
+  },
+  variants: {
+    extend: {},
+  },
+  plugins: [],
+}
