diff --git a/setup-vue-cli-2/src/App.vue b/setup-vue-cli-2/src/App.vue
index 55df315..b9ac61a 100644
--- a/setup-vue-cli-2/src/App.vue
+++ b/setup-vue-cli-2/src/App.vue
@@ -18,11 +18,6 @@ export default {
 
 <style>
 #app {
-  font-family: Avenir, Helvetica, Arial, sans-serif;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-  text-align: center;
-  color: #2c3e50;
   margin-top: 60px;
 }
 </style>
diff --git a/setup-vue-cli-2/src/main.css b/setup-vue-cli-2/src/main.css
new file mode 100644
index 0000000..110b5e1
--- /dev/null
+++ b/setup-vue-cli-2/src/main.css
@@ -0,0 +1,7 @@
+body {
+  font-family: Avenir, Helvetica, Arial, sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  text-align: center;
+  color: #2c3e50;
+}
diff --git a/setup-vue-cli-2/src/main.js b/setup-vue-cli-2/src/main.js
index 63eb05f..1d9028a 100644
--- a/setup-vue-cli-2/src/main.js
+++ b/setup-vue-cli-2/src/main.js
@@ -1,5 +1,6 @@
 import Vue from 'vue'
 import App from './App.vue'
+import './main.css'
 
 Vue.config.productionTip = false
 
