diff --git a/setup-vue-cli-2/.gitignore b/setup-vue-cli-2/.gitignore
index 403adbc..372a13f 100644
--- a/setup-vue-cli-2/.gitignore
+++ b/setup-vue-cli-2/.gitignore
@@ -21,3 +21,6 @@ pnpm-debug.log*
 *.njsproj
 *.sln
 *.sw?
+
+.nyc_output
+coverage
diff --git a/setup-vue-cli-2/.nycrc b/setup-vue-cli-2/.nycrc
new file mode 100644
index 0000000..3a40283
--- /dev/null
+++ b/setup-vue-cli-2/.nycrc
@@ -0,0 +1,13 @@
+{
+  "include": [
+    "src/**/*.{js,vue}"
+  ],
+  "exclude": [
+    "src/**/*.cy.js"
+  ],
+  "extension": [
+    ".js",
+    ".vue"
+  ],
+  "all": true
+}
\ No newline at end of file
diff --git a/setup-vue-cli-2/babel.config.js b/setup-vue-cli-2/babel.config.js
index e955840..f4b8fdb 100644
--- a/setup-vue-cli-2/babel.config.js
+++ b/setup-vue-cli-2/babel.config.js
@@ -1,5 +1,12 @@
-module.exports = {
-  presets: [
-    '@vue/cli-plugin-babel/preset'
-  ]
+const plugins = []
+
+if (process.env.CYPRESS) {
+  plugins.push('istanbul')
 }
+
+module.exports = {
+   presets: [
+     '@vue/cli-plugin-babel/preset'
+  ],
+  plugins
+}
\ No newline at end of file
diff --git a/setup-vue-cli-2/cypress.config.js b/setup-vue-cli-2/cypress.config.js
index eabdc78..6c3dd60 100644
--- a/setup-vue-cli-2/cypress.config.js
+++ b/setup-vue-cli-2/cypress.config.js
@@ -1,11 +1,20 @@
 const { defineConfig } = require('cypress')
 const { devServer } = require('@cypress/webpack-dev-server')
 const webpackConfig = require('@vue/cli-service/webpack.config')
+const codeCoverageTask = require("@cypress/code-coverage/task")

 module.exports = defineConfig({
+  coverage: true,
+  codeCoverage: {
+    exclude: "cypress/**/*.*",
+  },
   // Component testing, JavaScript, Vue CLI, Webpack
   component: {
     devServer,
-    devServerConfig: { webpackConfig }
+    devServerConfig: { webpackConfig },
+    setupNodeEvents(on, config) {
+      codeCoverageTask(on, config);
+      return config;
+    },
   },
 })
\ No newline at end of file
diff --git a/setup-vue-cli-2/cypress/support/component.js b/setup-vue-cli-2/cypress/support/component.js
index 7e654cd..a5eff01 100644
--- a/setup-vue-cli-2/cypress/support/component.js
+++ b/setup-vue-cli-2/cypress/support/component.js
@@ -2,4 +2,6 @@
 import './commands'

 // Ensure global app styles are loaded:
-import '../../src/main.css'
\ No newline at end of file
+import '../../src/main.css'
+
+import '@cypress/code-coverage/support'
\ No newline at end of file
